src/Packages/RetroMole.Render.Veldrid/Controller.cs

```diff
@@ -537,114 +524,108 @@ public void CreateDeviceResources()
         {
-            var factory = _gd.ResourceFactory;
-            var t = factory.CreateTexture(new TextureDescription(
+            Texture t = _gd.ResourceFactory.CreateTexture(new TextureDescription(
                 (uint)Texture.Width,
                 (uint)Texture.Height,
                 1,1,1,
                 PixelFormat.B8_G8_R8_A8_UNorm,
                 TextureUsage.Sampled,
                 TextureType.Texture2D
-            ));
+            ));    
             
-            _gd.UpdateTexture<byte>(t, Texture.Data, 0, 0, 0, (uint)Texture.Width, (uint)Texture.Height, 1, 0, 0);
+            _gd.UpdateTexture(t, Texture.Pixels, 0, 0, 0, (uint)Texture.Width, (uint)Texture.Height, 1, 0, 0);
 
-            if (!_autoViewsByTexture.TryGetValue(t, out TextureView textureView))
-            {
-                textureView = factory.CreateTextureView(t);
-                _autoViewsByTexture.Add(t, textureView);
-                _ownedResources.Add(textureView);
-            }
+            var tv = _gd.ResourceFactory.CreateTextureView(t);
+            var rs = _gd.ResourceFactory.CreateResourceSet(new(
+                _textRL,
+                tv
+            ));
 
-            if (!_setsByView.TryGetValue(textureView, out ResourceSetInfo rsi))
-            {
-                ResourceSet resourceSet = factory.CreateResourceSet(new ResourceSetDescription(_textureLayout, textureView));
-                rsi = new ResourceSetInfo(GetNextImGuiBindingID(), resourceSet);
+            var id = GetNextImGuiBindingID();
 
-                _setsByView.Add(textureView, rsi);
-                _viewsById.Add(rsi.ImGuiBinding, rsi);
-                _ownedResources.Add(resourceSet);
-            }
+            Textures.Add(new() {ID = id, Texture = rs});
 
-            return rsi.ImGuiBinding;
+            return id;
         }
-
-        private IntPtr GetNextImGuiBindingID()
+        
+        public override IntPtr UpdateTexture(IntPtr ID, Core.Interfaces.Texture texture)
         {
-            int newID = _lastAssignedID++;
-            return (IntPtr)newID;
-        }
+            Texture t = (Texture)Textures.First(t => t.ID == texture.ID).Texture;
 
-        public ResourceSet GetImageResourceSet(IntPtr imGuiBinding)
-        {
-            if (!_viewsById.TryGetValue(imGuiBinding, out ResourceSetInfo tvi))
-            {
-                throw new InvalidOperationException("No registered ImGui binding with id " + imGuiBinding.ToString());
-            }
+            _gd.UpdateTexture(t, texture.Pixels, 0, 0, 0, (uint)texture.Width, (uint)texture.Height, 1, 0, 0);
+            Textures.Remove(Textures.First(t => t.ID == texture.ID));
 
-            return tvi.ResourceSet;
+            var tv = _gd.ResourceFactory.CreateTextureView(t);
+            var rs = _gd.ResourceFactory.CreateResourceSet(new(
+                _textRL,
+                tv
+            ));
+
+            Textures.Add(new() {ID = texture.ID, Texture = rs});

```

Comment:
Again, updating textures this way is cumbersome, we need to switch back to linked dicts